build.gradle

```diff
@@ -123,7 +123,7 @@ dependencies {
     // Using the shaded version to avoid conflicts between its protobuf dependency
     // and that of Hadoop/Spark (either the one we reference explicitly, or the one
     // provided by dataproc).
-    compile 'com.google.cloud:google-cloud-nio:0.23.1-alpha:shaded'
+    compile 'com.google.cloud:google-cloud-nio:0.20.4-alpha-20170727.190814-1:shaded'
```

Comment:
Yes, I'm assuming this was change was unintentional. It, along with the corresponding change in CommandLineProgram, should both be removed from this PR.

Reply:
Shows this way because I based my PR on Louis' reversion.
@cmnbroad -- I included Louis' reversion so that I had code that worked on Spark.  If I eliminate these changes and rebase on master will I have something that works?

Reply:
@tedsharpe We just talked about this in the engine team meeting and we've merged Louis' reversion, so it should be in master now.